Loop/Managers/Alerts/AlertStore.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  PersistentDeviceAlertLog.swift
+//  Loop
+//
+//  Created by Rick Pasetto on 5/11/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import CoreData
+import LoopKit
+
+public class AlertStore {
+    
+    public enum AlertStoreError: Error {
+        case notFound
+    }
+       
+    private let managedObjectContext: NSManagedObjectContext
+
+    private let persistentContainer: NSPersistentContainer
+        
+    private let log = DiagnosticLog(category: "AlertStore")
+
+    private let dataAccessQueue = DispatchQueue(label: "com.loop.AlertStore.dataAccessQueue", qos: .utility)
+    
+    public init(storageFileURL: URL? = nil) {
+        managedObjectContext = NSManagedObjectContext(concurrencyType: .privateQueueConcurrencyType)
+        managedObjectContext.mergePolicy = NSMergeByPropertyStoreTrumpMergePolicy
+        managedObjectContext.automaticallyMergesChangesFromParent = true
+
+        let storeDescription = NSPersistentStoreDescription()
+        if let storageFileURL = storageFileURL {
+            storeDescription.url = storageFileURL
+            print("AlertStore: \(storageFileURL)")
+        } else {
+            storeDescription.type = NSInMemoryStoreType
+        }
+        persistentContainer = NSPersistentContainer(name: "AlertStore")
+        persistentContainer.persistentStoreDescriptions = [storeDescription]
+        persistentContainer.loadPersistentStores { description, error in
+            if let error = error {
+                fatalError("Unable to load persistent stores: \(error)")
+            }
+        }
+        managedObjectContext.persistentStoreCoordinator = persistentContainer.persistentStoreCoordinator
+    }
+}
+
+// MARK: Alert Recording
+
+extension AlertStore {
+    
+    public func recordIssued(alert: DeviceAlert, at timestamp: Date = Date(), completion: ((Result<Void, Error>) -> Void)? = nil) {
+        dataAccessQueue.async {
+            self.managedObjectContext.perform {
+                _ = StoredAlert(from: alert, context: self.managedObjectContext, issuedTimestamp: timestamp)
+                do {
+                    try self.managedObjectContext.save()
+                    self.log.default("Recorded alert: %{public}@", alert.identifier.value)
+                    completion?(.success)
+                } catch {
+                    self.log.error("Could not store alert: %{public}@", String(describing: error))
+                    completion?(.failure(error))
+                }
+            }
+        }
+    }
+    
+    private func lookupLatest(identifier: DeviceAlert.Identifier, completion: @escaping (Result<StoredAlert?, Error>) -> Void) {
+        managedObjectContext.perform {
+            let predicate = NSPredicate(format: "identifier = %@", identifier.value)
```

Comment:
[nit] I prefer `==` over `=` so it reads more like code, but whatever your preference is.

Reply:
you mean in the predicate?  Sure, that sounds like a good rule. 👍